components/nodemanager-service/pgdb/node-managers-process-node.go

```diff
@@ -86,9 +168,47 @@ func (db *DB) ProcessIncomingNode(node *manager.NodeMetadata) error {
 		}
 		return tx.tagNode(node.GetUuid(), tags)
 	})
+
+	return err
+}
+
+func (db *DB) getCurrentScanStatus(node *manager.NodeMetadata) (string, error) {
+	var status sql.NullString
+	err := db.QueryRow(sqlGetCurrentScanStatus, node.GetUuid(),
+		node.GetSourceId(), node.GetSourceRegion(), node.GetSourceAccountId()).Scan(&status)
+	switch {
+	case err == sql.ErrNoRows:
+		return "", nil
+	case err != nil:
+		return "", errors.Wrap(err, "unable to query for current status")
+	}
+	if !status.Valid {
+		return "", nil
+	}
+	statusString, err := strconv.Unquote(status.String)
 	if err != nil {
-		return err
+		return "", errors.Wrap(err, "unable to read status")
 	}
+	return statusString, nil
+}
```

Comment:
Unless I misunderstand this function and the one below should be able to use `db.SelectStr` to get the status in one line.

Reply:
ill give it a try!

Reply:
`	ProcessIncomingNode unable to get current scan status: unable to read status: sql: Scan error on column index 0, name "status": unsupported Scan, storing driver.Value type <nil> into type *string`
looks like selectstr isn't great here..

Reply:
Can you use `COALESCE` to always return a string?